### PR TITLE
Do not specialize 0/1 eagerly for storage offset

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #122636
* __->__ #113694

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by albanD — APPROVED** [review]
What do we expect the impact to be for current models that don't require this?

**Comment by ezyang** [comment]
In codegen, we expect that you have to do an extra offset computation to the data pointer before you do the actual compute. If the data pointer can be put in a register this should have no impact, but if we are dumb and inline it to all the spots, IDK, maybe CSE won't work?

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -2073,6 +2073,7 @@ def _create_symbolic_sizes_strides_storage_offset(
             TensorPropertySource(source, TensorProperty.STORAGE_OFFSET),
             dynamic_dim=DimDynamic.DYNAMIC,
             constraint_dim=None,
+            do_not_specialize_zero_one=True,
```
Comment (davidberard98) [inline torch/fx/experimental/symbolic_shapes.py:2076]: I think we also need `positive=False`

**Comment by albanD** [comment]
@ezyang  sounds good for runtime!
I meant in terms of compilation time due to the extra dynamic objects that need tracking. But I guess that's not a concern

**Comment by ezyang** [comment]
This is cursed lol

**Pull request closed without merging**